ensight_lib/include/ensightpart.h

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2016 Fraunhofer ITWM
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+
+#ifndef ENSIGHTPART_H
+#define ENSIGHTPART_H
+
+#include <map>
+#include <memory>
+#include <string>
+#include <QString>
+#include <QVector>
+
+#include "eigentypes.h"
+#include "ensightdef.h"
+
+class Bbox;
+class EnsightCell;
+class EnsightCellList;
+class EnsightVariable;
+
+/**
+ * @brief The EnsightPart class
+ */
+class EnsightPart
+{
+public:
+    EnsightPart(QString name, int id, int timesteps);
+    EnsightPart(std::string name, int id, int timesteps);
+    ~EnsightPart();
+
+
+    QString getName() const;
+    int getId() const;
+
+    /**
+     * @brief Deletes data for the given step
+     * @param step Timestep
+     */
+    void clean(int step);
+    /**
+     * @brief Set number of timesteps.
+     * @param timesteps
+     */
+    void setTimeSteps(int timesteps);
+    /**
+     * @brief Get the number of timesteps
+     */
+    int getNumberOfTimesteps() const;
+    /**
+     * @brief Sets the vertices and boundaries at a given timestep
+     * @param vertices A 3xN matrix containing N 3D vertices
+     * @param timestep Timestep
+     */
+    void setVertices(const Matx& vertices, int timestep);
+    void print() const;
```

Comment:
docstring (and probably should take an ostream)

Reply:
Fixed